### PR TITLE
FLINK-3529 Add template for pull requests

### PR DESCRIPTION
**Pull request by mliesenberg**
- added markdown as discussed on the mailing list
- added template to list of excluded files of rat plugin

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Thanks for contributing to Apache Flink, before you open your PR please kindly take into consideration the following check list.
```
Comment (vasia) [inline PULL_REQUEST_TEMPLATE.md]: I would split this in 2 sentences. i.e. Before...

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+Once you are sure, all items on the list can be checked, feel free to open your PR. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
```
Comment (vasia) [inline PULL_REQUEST_TEMPLATE.md]: Remove comma after sure.

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+In addition to going through the list, please provide a meaningful description of your changes.
+
+- [ ] General
+  - The pull request references the related JIRA issue
+  - The pull request addresses only one issue
+  - Each commit in the PR has a meaningful commit message
+
+- [ ] Documentation
+  - Documentation has been added for new functionality
+  - Old documentation updated affected by the pull request has been updated
```
Comment (vasia) [inline PULL_REQUEST_TEMPLATE.md]: Remove first occurrence of "updated".

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (mxm) [inline PULL_REQUEST_TEMPLATE.md]: Suggestion: Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.

File: PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mxm) [inline PULL_REQUEST_TEMPLATE.md]: Suggestion: If your changes take all of the items into account, feel free to open your pull request.

**Comment by hsaputra** [comment]
Should be a link from CONTRIBUTING page to this one.

**Comment by mliesenberg** [comment]
Thanks for the feedback, I will update the PR tonight.

**Comment by mliesenberg** [comment]
build failed because all jobs failed the `ClassLoaderITCase.testJobsWithCustomClassLoader` test.
I guess this is unrelated.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -8,4 +8,4 @@ To make the process smooth for the project *committers* (those who review and ac
 
 Please check out the [How to Contribute guide](http://flink.apache.org/how-to-contribute.html) to understand how contributions are made.
 There is also a list of [Coding Guidelines](http://flink.apache.org/coding-guidelines.html) that you should follow.
-
+Moreover, there's a [check list](PULL_REQUEST_TEMPLATE.md) of what to do before opening a pull request here
```
Comment (mxm) [inline CONTRIBUTING.md]: This is not a full sentence. How about: "For pull requests, there is a [check list](PULL_REQUEST_TEMPLATE.md) with criteria taken from the How to Contribute Guide and the Coding Guidelines."

File: PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+- [ ] Documentation
+  - Documentation has been added for new functionality
+  - Old documentation affected by the pull request has been updated
+  - JavaDoc for public methods has been added
+
+- [ ] Tests & Build
+  - Functionality added by the pull request is covered by tests
+  - `mvn clean verify` has been executed successfully locally
+  - or a Travis build has passed
```
Comment (mxm) [inline PULL_REQUEST_TEMPLATE.md]: Could you combine the last two in a single bullet point?

**Comment by mxm** [comment]
Yes, the test failures are unrelated. Looks good. Could you incorporate the last comments?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -7,5 +7,6 @@ To make the process smooth for the project *committers* (those who review and ac
 ## Contribution Guidelines
 
 Please check out the [How to Contribute guide](http://flink.apache.org/how-to-contribute.html) to understand how contributions are made.
+<<<<<<< Updated upstream
```
Comment (fhueske) [inline CONTRIBUTING.md]: Looks like a left-over of a merge/rebase conflict.

**Comment by fhueske** [comment]
I like the template! :-)

**Comment by mliesenberg** [comment]
thanks. removed the regression, squashed and rebased on master.

**Comment by fhueske** [comment]
Good to merge in my opinion. If there are no other comments on the template, I will merge the PR in a few days. 
Thanks again @mliesenberg!

**Comment by mxm** [comment]
Good to merge!

**Comment by hsaputra** [comment]
+1
LGTM

**Comment by mxm** [comment]
Will merge this if there are no further comments.

**Pull request closed without merging**